integration-libs/cdp/root/service/cdp-order-service.ts

```diff
@@ -4,19 +4,191 @@
+import { BehaviorSubject, Observable } from 'rxjs';
+import { product } from '../model/ImageDetail/product';
+import { finalOrder } from '../model/order/finalOrder';
+import { order } from '../model/orderDetail/order';
+import { Injectable } from '@angular/core';
+import { CdpOrderFacade } from '../facade/cdp-order-facade';
+import { returnOrder } from '../model/returnDetail/returnOrder';
+
+@Injectable()
+export class CdpOrderService implements CdpOrderFacade {
+  constructor(
+    private userIdService: UserIdService,
+    private cdpOrderAdapter: cdpOrderAdapter,
+    protected datePipe: CxDatePipe,
+    protected routing: RoutingService,
+    protected occEndpointsService: OccEndpointsService,
+    protected translation: TranslationService
+  ) {}
+
+  orderValue: finalOrder = { orders: [] };
+  totalPrice: number = 0;
+  totalItem: number[] = [];
+  orderDetail: Record<string, order> = {};
+  orderStatus: Record<string, Record<string, number>> = {};
+  orderImage: Record<string, product[]> = {};
+  userId: string;
+  tabTitleParam$ = new BehaviorSubject(0);
+  public loading$ = new BehaviorSubject<boolean>(true);
+  sortType: string;
+  obser$: Observable<finalOrder>;
+  returnObser$: Observable<returnOrder>;
+  orderData: order;
+  returnDate: Record<string, string>={};
+
+  public getOrder(page_size: number): Observable<finalOrder> {
+    this.obser$ = this.userIdService
+      .takeUserId()
+      .pipe(
+        switchMap((userId) => this.cdpOrderAdapter.getOrder(userId, page_size))
+      );
+
+    return this.obser$;
+  }
+
+  public fetchOrder(page: number, page_size: number): Observable<finalOrder> {
+    this.obser$ = this.userIdService
+      .takeUserId()
+      .pipe(
+        switchMap((userId) =>
+          this.cdpOrderAdapter.getOrderPerPage(userId, page_size, page)
+        )
+      );
+    return this.obser$;
+  }
+
+  public fetchReturn(): Record<string, string>{
+    this.returnObser$=this.userIdService.takeUserId().pipe(switchMap((userId) => this.cdpOrderAdapter.getRetunDetail(userId)));
+    //this.returnObser$=this.cdpOrderAdapter.getRetunDetail();
+    this.returnObser$.subscribe((returnSub)=>{
+      if(returnSub.returnRequest!=null){
+        returnSub.returnRequest.forEach((returnval)=>{
+          this.returnDate[returnval.ReefernceSDDocumentID]=returnval.creationTime;
+      });
+      }
+    });
+    return this.returnDate;
+  }
+
+  public async fetchOrderDetail(
+    finalResult: finalOrder
+  ): Promise<Record<string, order>> {
+    for (let orderval of finalResult.orders) {
+      await this.userIdService
+        .takeUserId()
+        .pipe(
+          mergeMap((userId) =>
+            this.cdpOrderAdapter.getOrderDetail(userId, orderval)
+          )
+        )
+        .toPromise()
+        .then((data) => {
+          this.orderDetail[orderval.code] = data;
+        });
+    }
+    return this.orderDetail;
+  }
+
+  public fetchOrderStatus(
```

Comment:
This method can be modified to be invoked immediately for each order-detail. In this way, we can avoid the `detail` object

We can discuss.